sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/resource/server/AADOAuth2OboAuthorizedClientRepository.java

```diff
@@ -0,0 +1,135 @@
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.core.AbstractOAuth2Token;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.server.resource.authentication.AbstractOAuth2TokenAuthenticationToken;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.net.MalformedURLException;
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+public class AADOAuth2OboAuthorizedClientRepository implements OAuth2AuthorizedClientRepository {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AADOAuth2OboAuthorizedClientRepository.class);
+
+    private final AzureClientRegistrationRepository azureClientRegistrationRepository;
+
+    private Map<String, ConfidentialClientApplication> confidentialClientApplicationMap = new HashMap<>();
+
+    public AADOAuth2OboAuthorizedClientRepository(AzureClientRegistrationRepository azureClientRegistrationRepository) {
+        this.azureClientRegistrationRepository = azureClientRegistrationRepository;
+        Iterator<ClientRegistration> iterator = azureClientRegistrationRepository.iterator();
+        while (iterator.hasNext()) {
+            ClientRegistration next = iterator.next();
+            this.confidentialClientApplicationMap.put(next.getClientId(), createApp(next));
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T extends OAuth2AuthorizedClient> T loadAuthorizedClient(String registrationId,
+                                                                     Authentication authentication,
+                                                                     HttpServletRequest request) {
+        try {
+            ClientRegistration clientRegistration = azureClientRegistrationRepository
+                .findByRegistrationId(registrationId);
+
+            AbstractOAuth2TokenAuthenticationToken<AbstractOAuth2Token> authenticationToken =
+                (AbstractOAuth2TokenAuthenticationToken)
+                    authentication;
+
+            String accessToken = authenticationToken.getToken().getTokenValue();
+
+            OnBehalfOfParameters parameters = OnBehalfOfParameters
+                .builder(clientRegistration.getScopes(), new UserAssertion(accessToken))
+                .build();
+
+            ConfidentialClientApplication clientApplication = confidentialClientApplicationMap.get(clientRegistration
+                .getClientId());
+
+            String oboAccessToken = clientApplication.acquireToken(parameters).get().accessToken();
```

Comment:
Hi, @sangonzal, @SomkaPe,
We are using msal to implement obo function in spring-framework.
Could you please help to review this PR?